dash-pipeline/bmv2/underlay.p4

```diff
@@ -36,6 +36,8 @@ control underlay(
     #endif // TARGET_BMV2_V1MODEL
 
     apply {
+#ifdef TARGET_BMV2_V1MODEL
```

Comment:
Won't this build on DPDK as well? If not, why not? The intent is to compile all DASH code for both targets. The `#ifdef`s are to handle target-specific code if it's needed, ideally there would be none at all.

Reply:
The standard_metadata in the control block has been wrapped inside ifdef TARGET_BMV2_V1MODEL.
We are using standard_metadata for routing and hence the ifdef is required

Reply:
Your response is incomplete. You didn't answer 

While P4DPDK uses a different nomenclature for metadata, it's handled by the `#ifdef` in https://github.com/AmithGspn/DASH/blob/64cbb0ae83f65b14469f8554ace942ca18c3a8d1/dash-pipeline/bmv2/dash_pipeline.p4#L19 and then the ports are handled slightly differently e.g. https://github.com/AmithGspn/DASH/blob/64cbb0ae83f65b14469f8554ace942ca18c3a8d1/dash-pipeline/bmv2/dash_pipeline.p4#L382> You should examine the code for anything like this affecting your enhancement.

The goal is to have the same "program" behavior for either target, even if certain architectural details differ. Those are handled with `#ifdefs`. So, you should call your `underlay()` regardless of target and handle the port setting appropriately.